### DISCUSSION TITLE
When to use Simulator.injectAction vs Simulator.registerAction?

### DISCUSSION THREAD

**Discussion by quekyj** [intel/rohd#342] [Q&A]
When should we use `Simulator.injectAction`? From the ROHD unit test, an example of of `injectAction` usage is

```dart
Simulator.registerAction(50, () async {
      Simulator.injectAction(() async {
        await Future<void>.delayed(const Duration(microseconds: 10));
        injectedActionExecuted = true;
      });
});
```
I am thinking if there any different with this:

```dart
Simulator.registerAction(50, () async {
        await Future<void>.delayed(const Duration(microseconds: 10));
        injectedActionExecuted = true;
});
```

**Comment by mkorbel1** [comment]
These unit tests are kind of weird because they are specifically testing the behavior of the simulator in specific situations, so they may be initially confusing as examples.

The documentation gives some explanation on the differences between [`registerAction`](https://intel.github.io/rohd/rohd/Simulator/registerAction.html) and [`injectAction`](https://intel.github.io/rohd/rohd/Simulator/injectAction.html).  The important difference is that `injectAction` will attempt to make the event happen in the *current* timestep if possible (via multiple tick's in one timestamp), whereas `registerAction` will set up an event to occur at a specific time in the simulation (not including the current time).
